MRG:  allow distributors to run custom init

matthew-brett opened the pull request:
Give hook to allow platform-specific installs to modify the
initialization of numpy.  Particular use-case is to allow check for SSE2
on Windows when shipping with ATLAS wheel.

matthew-brett committed:
ENH: platform-specific install hook to change init

Give hook to allow platform-specific installs to modify the
initialization of numpy.  Particular use-case is to allow check for SSE2
on Windows when shipping with ATLAS wheel.

njsmith commented:
I'd make it `import ._distributor_init` -- both for the clearer name, and because there's no reason to do `import *` and pollute the namespace. But otherwise I like the idea and had been thinking of proposing it too :-)

matthew-brett commented:
OK - I changed it as you suggested.

njsmith commented:
LGTM. But I'll leave this open for others to comment... in fact it might be useful to send an email to the list, and/or CC'ing our major known distributors like @cgohlke and Sandro.

rgommers commented:
@rkern's suggestion of  unconditional import of an empty ``numpy/_distributor_init.py`` seems to be slightly more optimal, but other than that +1

matthew-brett commented:
OK - modified as Robert suggested.

charris commented:
Needs mention in the 1.12 release notes

matthew-brett committed:
DOC: description of _distribution_init for release

Add description of ``numpy/_distribution_init.py`` file and init hook to
release notes for 1.12.0.

homu commented:
:umbrella: The latest upstream changes (presumably #7246) made this pull request unmergeable. Please resolve the merge conflicts.

charris commented:
Needs rebase also...

charris commented:
@matthew-brett Needs rebase, probably the release notes.

matthew-brett commented:
Done - thanks for the reminder.​
